inc/Utf8/Sort.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace dokuwiki\Utf8;
+
+/**
+ * DokuWiki sort functions
+ *
+ * When "intl" extension is available, all sorts are done using a collator.
+ * Otherwise, primitive PHP functions are called.
+ *
+ * The collator is created using the locale given in $conf['lang'].
+ * It always uses case insensitive "natural" ordering in its collation.
+ * The fallback solution uses the primitive PHP functions that return almost the same results
+ * when the input is text with only [A-Za-z0-9] characters.
+ *
+ * @license    GPL 2 (http://www.gnu.org/licenses/gpl.html)
+ * @author     Moisés Braga Ribeiro <moisesbr@gmail.com>
+ */
+class Sort
+{
+    /** @var \Collator[] language specific collators, usually only one */
+    protected static $collator;
+
+    /**
+     * Initialization of a collator using $conf['lang'] as the locale.
+     * The initialization is done only once, except when $reload is set to true.
+     * The collation takes "natural ordering" into account, that is, "page 2" is before "page 10".
+     *
+     * @param bool $reload Usually false; true forces collator re-creation
+     * @return \Collator Returns a configured collator or null if the collator cannot be created.
+     *
+     * @author Moisés Braga Ribeiro <moisesbr@gmail.com>
+     */
+    protected static function getCollator($reload = false)
+    {
+        global $conf;
+        $lc = $conf['lang'];
+
+        // check if intl extension is available
+        if (!class_exists('\Collator')) {
+            return null;
+        }
+
+        // load collator if not available yet
+        if ($reload || !isset(self::$collator[$lc])) {
+            $collator = \Collator::create($lc);
+            if (!isset($collator)) return null;
```

Comment:
INFO: this check is needed by the library docs, but in practice the library never returns null.